crates/diagnostics/src/items.rs

```diff
@@ -22,6 +23,17 @@ pub struct DiagnosticIndicator {
 
 impl Render for DiagnosticIndicator {
     fn render(&mut self, cx: &mut ViewContext<Self>) -> impl IntoElement {
+        if let Some(elements) = WorkspaceSettings::get_global(cx)
+            .status_bar
+            .clone()
+            .elements
+        {
+            // Now that we have elements, check cursor_position
+            if !elements.diagnostics.unwrap_or(true) {
+                // If cursor_position is true or None (defaulting to true)
+                return div();
+            }
+        }
```

Comment:
not a rust expert myself, unfortunately

Reply:
Should these be optional? Is that how other settings are implemented? If so, it feels like the fallback setting should come from a proper source in the case that they are `None` rather than dispersed `unwrap_or` calls.

Reply:
You are right, others generally aren't optional. They are just bool instead. 

I can make this change